xdsl/dialects/vector.py

```diff
@@ -846,6 +847,133 @@ def resolve_attrs(
 
         return resolved_mask, permutation_map, in_bounds
 
+    def is_broadcast_dim(self, dim: int) -> bool:
+        expr = self.permutation_map.data.results[dim]
+        if not isa(expr, AffineConstantExpr):
+            return False
+        return expr.value == 0
+
+    def has_broadcast_dim(self):
+        transfer_rank = len(self.permutation_map.data.results)
+
+        for dim in range(transfer_rank):
+            if self.is_broadcast_dim(dim):
+                return True
+
+        return False
```

Comment:
I find this a little easier to read, and it avoids the whole "get range then index" pattern that's much slower in Python than in cpp.
```suggestion
    def has_broadcast_dim(self):
        """
        Return "true" if at least one of the vector dimensions is a broadcasted dimension.
        """
        return any(
            isinstance(expr, AffineConstantExpr) and expr.value == 0
            for expr in self.permutation_map.data.results
        )
```